### ISSUE TITLE
Position margin level

### ISSUE THREAD

**Issue by gordsport** [vegaprotocol/frontend-monorepo#473]
As a user of console I want to see how close I am to being liquidated.

#### Acceptance criteria
- [x] Show estimated liquidation price
- [x] Show entry price
- [x] Show progress bar indicating how close you are to the liquidation price
- [x] Color should change using Intent to indicate 'danger' level using @witgaw calculations below

**Comment by gordsport** [comment]
@witgaw - can you please update on the merging of the python notebook regarding margin warnings so we can use this?

**Comment by campbellssource** [comment]
@PeteWilliams and @witgaw to discuss a good time/price point to start warning

**Comment by PeteWilliams** [comment]
Have talked to @witgaw who has proposed the following conditions should trigger a warning:

1.   `margin account balance` < `search level` + (`initial level - search level)/2` [we’re more than half way down between initial level and search level]
AND
  2.  `collateral account balance` < `initial level - search level` [the margin won’t be topped up to the desired level <-- they still don’t get closed out as long as `margin account balance` (after topup) > `search level`, but they are cutting it close so I think it warrants a warning.

**Comment by campbellssource** [comment]
Here is the workbook for calculating liquidation level: https://github.com/vegaprotocol/research/blob/master/notebooks/misc/Misc%20position%20metrics.ipynb

**Comment by PeteWilliams** [comment]
Have added initial designs for this:
https://www.figma.com/file/K49Nx7XfocShhHcwwQzQkm/Console-V2?node-id=2172%3A32829

When the position hits the condition [referenced above](https://github.com/vegaprotocol/frontend-monorepo/issues/473#issuecomment-1164290751), it's liquidation indicator bar should turn from white to orange (the 'warning' colour):
![image](https://user-images.githubusercontent.com/1325606/175934415-6e6e9f30-46ef-4416-a2df-6297927d6132.png)

We should also show a toast that alerts the user that this has happened:
![image](https://user-images.githubusercontent.com/1325606/175934633-623bcefd-ec93-4a8e-abc4-3aee18d876c2.png)

The toast would be triggered when this first happens (or when the user first logs in after this has happened, so long as the condition is still true), but we should also repeat the reminder as the user gets closer to liquidation. Happy to discuss metrics we might use to do this. The above metric alerts the user when they are more than 50% between initial level and search level - one method might be to alert them again when they are at 60%, 70%, 80%, 90% and 95%? Though in a fast-moving market where the market gaps, where they are all hit at once, we would just show a single toast for the highest number. Likewise if they log in having breached multiple barriers.

**Comment by gordsport** [comment]
**Scope discussion:** Toast part of this issue should be deferred to a new ticket in the future.

**Comment by PeteWilliams** [comment]
.

**Comment by bglownia** [comment]
TODO:
- Add positions table tests libs\positions\src\lib\positions-table.spec.tsx
- verify getMetrics against https://github.com/vegaprotocol/research/blob/master/notebooks/misc/Misc%20position%20metrics.ipynb
- addd more unit test with corner cases libs\positions\src\lib\positions-metrics-data-provider.spec.ts
- Fix/adjust E2E tests
- maybe replace libs\positions\src\lib\positions-data-provider.ts with positions-metrics-data-provider.ts

**Comment by bglownia** [comment]
Blocked by 
https://github.com/vegaprotocol/data-node/pull/830
being deployed to test net
